File: Java Old Code/Class12/src/Matrix.java
import java.util.*;
public class Matrix {
    int Octal(int num)
    {
        int oct = 0;
        int i = 0;
        while (num!=0)
        {
            oct += ((int)Math.pow(8,i))*(num%10);
            i++;
            num /=10;
        }
        return oct;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the Number of rows");
        int M = sc.nextInt();
        System.out.println("Enter the Number of columns");
        int N = sc.nextInt();
        int arr[][] = new int[M][N];
        Matrix ob = new Matrix();
        if(!(M>0&&M<10&&N>2&&N<6))
        {
            System.out.println("Wrong Input");
            System.exit(0);
        }
        for(int i=0;i<M;i++)
        {
            for(int j=0;j<N;j++)
            {
                System.out.println("Enter Element in Row:"+(i+1)+" Column:-"+(j+1));
                int n=0;
                while(true)
                {
                    n = sc.nextInt();
                    if(n>=0&&n<=7)
                        break;
                    System.out.println("Invalid Input");
                }
                arr[i][j] = n;
            }
        }
        for(int i=0;i<M;i++)
        {
            int n = 0;
            for(int j=0;j<N;j++) {
                System.out.print(arr[i][j]);
                n = n * 10 + arr[i][j];
            }
            System.out.println("\tInteger Equivalent:- "+ob.Octal(n));
        }
    }
}

File: C_Programs/Matrix.c
#include <stdio.h>
int main()
{
    int n;
    printf("Enter n ");
    scanf("%d",&n);
    int arr[n][n];
    for(int i=0;i<n;i++)
    {
        for(int j=0;j<n;j++)
        {
            printf("Enter an element in %d row and %d column ",i+1,j+1);
            scanf("%d",&arr[i][j]);
        }
    }
    int flag = 0;
    for(int i=0;i<n;i++)
    {
        for(int j=0;j<n;j++)
        {
            if(i<j && arr[i][j]!=0)
                flag = 1;
        }
    }
    if (flag==1)
        printf("Its not an Upper Triangular Matrix");
    else
        printf("It is an Upper Triangular Matrix");
    return 0;
}